### PR TITLE
Version Control Policy

### PR DESCRIPTION
**Pull request by 8ix**
This has been produced as part of a compliance for ISO 27001 and states how we work with Git Repositories at 383 project. Very high level security and im happy to ajust the methology and even link out to seperate Code of conducts for departments if you want.  

Please can you put some comments where you want changes otherwise approve for me so that I know I have your consent.

Many thanks 
Andrew

### CODE REVIEW

**Review by Maddock87 — DISMISSED** [review]
All good with me

File: lifecycle/build/version-control-policy.md [hunk 1]
```diff
@@ -1 +1,46 @@
...
+- 3.3 We will follow the [Gitflow methodology](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) for code development.
+- 3.4 All changes to the code will be made via branches, with pull requests used for code review and approval.
+- 3.5 Commits to the master/main branch will only be made after a successful code review and approval process.
+- 3.6 Commits must include a descriptive commit message, detailing the changes made and the purpose of the changes.
+- 3.7 All code changes will be tracked and recorded in Github, with each commit linked to an issue or feature.
+- 3.8 Code releases will be tagged and versioned in Github, with each release linked to a milestone or sprint.
+
+## 4. Access and Security
+- 4.1 Access to code repositories will be granted on a need-to-know basis, with least privilege access granted to all users.
+- 4.2 Access to code repositories will be managed through [383 Project Organisation Github Account](https://github.com/383Project/engineering-code-of-conduct), with user access rights reviewed and updated on a regular basis.
```
[the remark this answers is not in the document]
Comment (8ix) [reply lifecycle/build/version-control-policy.md:23]: Updated URL

File: lifecycle/build/version-control-policy.md [hunk 2]
```diff
@@ -1 +1,46 @@
...
+- 4.3 Authentication for Github access will be via multi-factor authentication (MFA) to ensure the security of the code repositories.
+- 4.4 All code repositories will be backed up regularly, with backups stored securely in a separate location from the primary repositories.
+- 4.5 Code repositories will be protected from unauthorized access or modification by implementing security controls that is overseen by the head of engineering and the information security manager. 
+- 4.6 To aquire access to a repository you will be required to put a request through to the controls administrator or your line manager to arrange the appropriate level of access.
```
[the remark this answers is not in the document]
Comment (8ix) [reply lifecycle/build/version-control-policy.md:27]: Added further notes and linked to department access control list

File: lifecycle/build/version-control-policy.md [hunk 3]
```diff
@@ -1 +1,46 @@
...
+- 5.2 Third-party access to code repositories will be granted on a need-to-know basis, with the level of access granted determined by our organization.
+- 5.3 Third-party access to code repositories will be managed through Github, with user access rights reviewed and updated on a regular basis.
+- 5.4 Third-party access to code repositories will be granted only after the completion of a security and risk assessment, and any necessary security controls will be implemented to protect our code repositories.
+- 5.5 Any third-party code changes will be tracked and recorded in Github, with each commit linked to an issue or feature.
+- 5.6 We will require third-party code to be released and versioned in accordance with our version control policy and procedures.
+- 5.7 Third-party access will be terminated and the relevent control documents will be updated at the point the third part relationship is terminated.
+- 5.8 383 Project is not responsible for the controls, security and access of external repositories provided to us that are administrated by third parties or partnering agencies.
```
[the remark this answers is not in the document]
Comment (8ix) [reply lifecycle/build/version-control-policy.md:37]: Yep, added this clarification of third parties to the doc

File: lifecycle/build/version-control-policy.md [hunk 4]
```diff
@@ -1 +1,46 @@
...
+
+## 6. Compliance and Audit
+- 6.1 Compliance with this policy will be monitored by the Engineering department, with regular control audits conducted to ensure that the policy is being followed and the correct level of access is appropriate.
+- 6.2 Any non-compliance with this policy will be reported to the Engineering department, and appropriate action will be taken to rectify the issue.
```
[the remark this answers is not in the document]
Comment (8ix) [reply lifecycle/build/version-control-policy.md:41]: Yep,  added note for this

**Pull request merged**